File: leetcode/53.cpp
// Probelm 53 in leetcode
// Using Kadane's algorithm for get Maximum sub array

#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = nums[0], maxValue = nums[0], ret = nums[0];
        
        for(int i = 1; i < nums.size(); i++){
            maxValue += nums[i];
            maxValue = max(maxValue, nums[i]);
            if(maxValue > ret) ret = maxValue;
        }
        
        return ret;
    }
};

File: leetcode/53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        maxVal = -1001
        for i in range(len(nums)):
            for j in range(i + 1, len(nums) + 1):
                s = sum(nums[i:j])
                maxVal = max(s, maxVal)
        return maxVal